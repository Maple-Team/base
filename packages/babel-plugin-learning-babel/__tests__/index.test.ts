import { getTransformCode } from './test-help'

describe('cases', () => {
  it('case1', () => {
    const sourceCode = `
    export const str = 'VariableDeclarator测试文案'
    `
    const result = getTransformCode(sourceCode, 'filename.tsx')
    expect(result?.code).toMatchSnapshot()
  })
  it('case2', () => {
    const sourceCode = `
  export const Component = () => {
    return <div>{'jsxExpressionContainer测试文案'}</div>
  }
  `
    const result = getTransformCode(sourceCode, 'filename.tsx')
    expect(result?.code).toMatchSnapshot()
  })
  it('case3', () => {
    const sourceCode = `
  export const Component = () => {
    return <g id="g-jsx属性"></g>
  }
  `
    const result = getTransformCode(sourceCode, 'filename.tsx')
    expect(result?.code).toMatchSnapshot()
  })
  it('case4', () => {
    const sourceCode = `
  export const Component = () => {
    return <div class="div-类名"></div>
  }
  `
    const result = getTransformCode(sourceCode, 'filename.tsx')
    expect(result?.code).toMatchSnapshot()
  })
})
