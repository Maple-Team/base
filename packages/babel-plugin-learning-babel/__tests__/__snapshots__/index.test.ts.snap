// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cases case1 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.str = void 0;
var str = exports.str = 'VariableDeclarator测试文案';"
`;

exports[`cases case2 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var Component = exports.Component = function Component() {
  return /*#__PURE__*/React.createElement(\\"div\\", null, 'jsxExpressionContainer测试文案');
};"
`;

exports[`cases case3 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var Component = exports.Component = function Component() {
  return /*#__PURE__*/React.createElement(\\"g\\", null);
};"
`;

exports[`cases case4 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.Component = void 0;
var Component = exports.Component = function Component() {
  return /*#__PURE__*/React.createElement(\\"div\\", {
    \\"class\\": \\"div-\\\\u7C7B\\\\u540D\\"
  });
};"
`;
